includes/class-event.php

```diff
@@ -354,4 +359,33 @@ private function get_refreshed_schedule_interval() {
 		// If we couldn't get from schedule (was removed), use whatever was saved already.
 		return $this->interval;
 	}
+
+	private function args_array_is_reasonably_sized(): bool {
+		// We aim to cache queries w/ up to 500 events.
+		$max_events_per_page = 500;
+
+		// A compressed db row of an event is around 300 bytes.
+		$db_row_size_of_normal_event = 300;
+
+		// Note: Memcache can only cache up to 1mb values, after compression.
+		$default_storage_needed_for_events = $db_row_size_of_normal_event * $max_events_per_page;
+		$reasonable_size = ( MB_IN_BYTES - $default_storage_needed_for_events ) / $max_events_per_page;
```

Comment:
Would this simplification still get you the size you want?

```suggestion
		// A compressed db row of an event is around 300 bytes.
		$db_row_size_of_normal_event = 300;

		// Note: Memcache can only cache up to 1mb values, after compression.
		$reasonable_size = ( MB_IN_BYTES / $max_events_per_page ) - $db_row_size_of_normal_event;
```
Is $db_row_size_of_normal_event kinda the amount you need to store the stuff around the event excluding the actual arguments?

Reply:
Right, it's roughly the size of a normal event row w/ an empty args array, after compression. I rounded up a little for some extra padding.